tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/getArtifactFilesCli.ts

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { getArtifactFilesInput, GetArtifactFilesInput } from './cliSchema/getArtifactFilesCliConfig';
+import {
+    GenerateAndBuildOutput,
+    getGenerateAndBuildOutput,
+    logger,
+    requireJsonc,
+    SDK,
+} from '@azure-tools/sdk-generation-lib';
+import { getFileListInPackageFolder } from './utils/git';
+
+async function getSourceCode(generateAndBuildOutput: GenerateAndBuildOutput, language: string, artifactDir: string) {
+    for (const p of generateAndBuildOutput.packages) {
+        const result = p.result;
+        if (result === 'failed') {
+            logger.warn(`Build ${p.packageName} failed, skipped it`);
+            continue;
+        }
+        const packageName = p.packageName;
+        const packageFolder = p.packageFolder;
+
+        if (packageFolder && fs.existsSync(packageFolder)) {
+            for (const filePath of getFileListInPackageFolder(packageFolder)) {
+                if (fs.existsSync(path.join(packageFolder, filePath))) {
+                    const fileDir = path.dirname(filePath);
+                    fs.mkdirSync(`${artifactDir}/${language}/${packageName}/${fileDir}`, { recursive: true });
```

Comment:
check exist before create the dir

It is not good way to implement in this way. Too many call to create an dir. Need to implement in the recursive function call as following:
function copyFolderRecursiveSync( source, target ) {
    var files = [];

    // Check if folder needs to be created or integrated
    var targetFolder = path.join( target, path.basename( source ) );
    if ( !fs.existsSync( targetFolder ) ) {
        fs.mkdirSync( targetFolder );
    }

    // Copy
    if ( fs.lstatSync( source ).isDirectory() ) {
        files = fs.readdirSync( source );
        files.forEach( function ( file ) {
            var curSource = path.join( source, file );
            if ( fs.lstatSync( curSource ).isDirectory() ) {
                copyFolderRecursiveSync( curSource, targetFolder );
            } else {
                copyFileSync( curSource, targetFolder );
            }
        } );
    }
}

Reply:
from now, use '.gitignore' to filter files and for each file's create it's dir and copy it maybe not efficient, will improve in future version